stage1: `zig help` prints usage info to stdout

jayschwa opened the pull request:
This will make it easier to do things like `zig help | grep something`.

When an invalid command or option is used, a short message recommending `zig help` is printed.
This prevents the actual error message from being drowned out by the large usage text.

andrewrk commented on src/main.cpp:
```diff
@@ -1021,6 +1037,7 @@ int main(int argc, char **argv) {
     case CmdInvalid:
-        return usage(arg0);
+        print_usage(stdout, arg0);
+        return EXIT_SUCCESS;
```
If we're going to return `EXIT_SUCCESS`, let's add a `help` command which prints to stdout and returns EXIT_SUCCESS. An invalid command should print to stderr and return `EXIT_FAILURE`.

jayschwa replied:
While the case is for `CmdInvalid`, this branch only executes when `zig` is called without arguments. I considered renaming it to `CmdNone` for that reason. Invalid commands are handled here: https://github.com/ziglang/zig/blob/master/src/main.cpp#L705-L706

I will add a `help` sub-command since I think it's good practice for any CLI tool. Perhaps, bare `zig` or any other argument error should print an abbreviated usage message pointing the user toward `zig help`?

andrewrk commented on src/main.cpp:
```diff
@@ -719,16 +736,19 @@ int main(int argc, char **argv) {
                     fprintf(stderr, "Unexpected extra parameter: %s\n", arg);
-                    return usage(arg0);
-                case CmdInvalid:
+                    fprintf(stderr, ZIG_HELP_NOTICE, arg0);
```
How about  `return badUsage(arg0)` which does `fprintf(stderr, ZIG_HELP_NOTICE, arg0);` and `return EXIT_FAILURE;`? This would match the original code, except with your new behavior. Then you don't have to have the `%s` in a separate line than the argument.

andrewrk reviewed (changes requested):
almost there

andrewrk commented:
Could you squash your commits?

jayschwa committed:
stage1: Added `zig help` to show usage on stdout

This will make it easier to do things like `zig help | grep something`.

Invalid arguments will now display a short notice for `zig help`
instead of showing the full usage information. This will make it easier
to see the actual error.

jayschwa commented:
I've addressed the latest round of comments.

andrewrk commented:
Thanks!
